common/src/main/java/io/netty/util/ResourceLeakDetector.java

```diff
@@ -460,47 +453,45 @@ public String toString() {
             "io.netty.buffer.AdvancedLeakAwareByteBuf.recordLeakNonRefCountingOperation("
     };
 
-    static String newRecord(Object hint, int recordsToSkip) {
-        StringBuilder buf = new StringBuilder(4096);
+    private static final class Record extends Throwable {
+        private final String hintString;
+        private Record next;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this "next" field final. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1165)

Reply:
false positive

Reply:
actually I think you want to make this field package private.